airbyte-integrations/connectors/source-shopify/unit_tests/graphql_bulk/test_job.py

```diff
@@ -167,6 +158,46 @@ def test_job_check(mocker, request, requests_mock, job_response, auth_config, er
             requests_mock.get(job_result_url, json=request.getfixturevalue(job_response))
         result = stream.job_manager.job_check(test_job_status_response)
         assert expected == result
+        
+
+@pytest.mark.parametrize(
+    "job_response, error_type, patch_healthcheck, expected",
+    [
+        (
+            "bulk_successful_response_with_errors", 
+            ShopifyBulkExceptions.BulkJobUnknownError, 
+            True, 
+            "Could not validate the status of the BULK Job",
+        ),
+    ],
+    ids=[
+        "success with errors (edge)",
+    ],
+)
+def test_one_time_retry_job_check(mocker, request, requests_mock, job_response, auth_config, error_type, patch_healthcheck, expected) -> None:
+    stream = MetafieldOrders(auth_config)
+    # modify the sleep time for the test
+    stream.job_manager.concurrent_max_retry = 1
+    stream.job_manager.concurrent_interval_sec = 1
+    stream.job_manager.job_check_interval_sec = 1
+    # get job_id from FIXTURE
+    job_id = request.getfixturevalue(job_response).get("data", {}).get("node", {}).get("id")
+    # patching the method to get the right ID checks
+    if job_id:
+        mocker.patch("source_shopify.shopify_graphql.bulk.job.ShopifyBulkManager.job_get_id", value=job_id)
+    if patch_healthcheck:
+        mocker.patch("source_shopify.shopify_graphql.bulk.job.ShopifyBulkManager.job_healthcheck", value=job_response)
+    # mocking the response for STATUS CHECKS
+    requests_mock.post(stream.job_manager.base_url, json=request.getfixturevalue(job_response))
+    test_job_status_response = requests.post(stream.job_manager.base_url)
+    
+    with pytest.raises(error_type) as error:
+        stream.job_manager.job_check(test_job_status_response)
+        # check the flag was set to `True`, meaning `Retried`
+        assert stream.job_manager.one_time_error_retried
```

Comment:
With requests_mock, can we assert the call_count instead? It feels that that's the behavior we want. The flag `one_time_error_retried` does not seem to indicate actual behavior

Reply:
Done.